.konflux/scripts/render_snapshot.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import datetime
+import json
+import os
+import re
+
+
+def parse_image_refs(image_refs):
+    return json.loads(image_refs)
+
+
+def validate_component(component):
+    assert (
+        component["name"] != ""
+        and component["containerImage"] != ""
+        and component["revision"] != ""
+        and component["repository"] != ""
+    ), "Component must have component name, ref, revision and repository set. Check container image labels."
```

Comment:
I feel like the recommendation to check container labels is unfortunately very much misplaced here.
This could be countered by adding information about which exact labels are to check for which image but that's even worse.

Imagine this task is moved to a https://github.com/stackrox/konflux-tasks repo and so it's abstracted away from how it's used in the pipeline and what the pipeline does to extract that information, yet this task would give the information about what some other task should do.
Spaghetti.

I think, it may be useful to keep this emptiness check here but validating the presence of labels must be a business of the `wait-for-image` task.